src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -39,22 +39,25 @@ import { selectAccountIdsByChainIdFilter } from 'state/slices/portfolioSlice/sel
 import { selectFeatureFlag } from 'state/slices/selectors'
 import { store, useAppSelector } from 'state/store'
 
-type WalletDeviceSupportsChainArgs = {
+type WalletSupportsChainArgs = {
   isSnapInstalled: boolean | null
   chainId: ChainId
   wallet: HDWallet | null
+  // The connected account ids to check against. If set to false, the currently connected account
+  // ids will not be checked (used for initial boot)
+  checkConnectedAccountIds: AccountId[] | false
```

Comment:
I opted to combine `chainAccountIds` with the flag triggering the check of those account IDs, because otherwise we'd have to do the following:
* in the case we DO want to check the account IDs, pass in the account IDs and set the flag to true
* in the case we DONT want to check the account IDs (e.g on fresh boot), pass in an empty array for the account IDs (since we dont have them) and set the flag to false

instead it makes way more sense to pass in the account ID when you have them, or set the param to false to explicitly say you need to skip the check.

Reply:
Love this one - much much better than having two args, and to pass `connectedAccountIds` when we don't need it 🧠